ecs-cli/modules/cli/local/up_app.go

```diff
@@ -14,18 +14,182 @@
+		WithDecryption: aws.Bool(true),
+	})
+	if err != nil {
+		return "", errors.Wrapf(err, "Failed to retrieve decrypted secret from %s due to %v", arnOrName, err)
+	}
+	return *val.Parameter.Value, nil
+}
+
+// DecryptSecret returns the decrypter secret value from Secrets Manager.
+func (d *secretsManagerDecrypter) DecryptSecret(arn string) (string, error) {
+	val, err := d.client.GetSecretValue(&secretsmanager.GetSecretValueInput{
+		SecretId: aws.String(arn),
+	})
+	if err != nil {
+		return "", errors.Wrapf(err, "Failed to retrieve decrypted secret from %s due to %v", arn, err)
+	}
+	return *val.SecretString, nil
+}
+
 // Up creates a Compose file from an ECS task definition and runs it locally.
 //
 // The Amazon ECS Local Endpoints container needs to be running already for any local ECS task to work
 // (see https://github.com/awslabs/amazon-ecs-local-container-endpoints).
 // If the container is not running, this command creates a new network for all local ECS tasks to join
 // and communicate with the Amazon ECS Local Endpoints container.
 func Up(c *cli.Context) {
-	client := docker.NewClient()
-	network.Setup(client)
+	// TODO When we don't provide any flags, Create() should just check if a "./docker-compose.local.yml" already exists.
+	// If so then Create() should do nothing else, otherwise it should error.
+	Create(c)
+
+	network.Setup(docker.NewClient())
+
+	config := readComposeFile(c)
+	secrets := readSecrets(config)
+	envVars := decryptSecrets(secrets)
+	upComposeFile(config, envVars)
+}
+
+func readComposeFile(c *cli.Context) *composeV3.Config {
+	filename := localproject.LocalOutDefaultFileName
+	if c.String(flags.LocalOutputFlag) != "" {
+		filename = c.String(flags.LocalOutputFlag)
+	}
+	config, err := converter.UnmarshalComposeFile(filename)
+	if err != nil {
+		logrus.Fatalf("Failed to unmarshal Compose file %s due to %v", filename, err)
+	}
+	return config
+}
+
+func readSecrets(config *composeV3.Config) []*secrets.ContainerSecret {
+	var containerSecrets []*secrets.ContainerSecret
+	for _, service := range config.Services {
+		for label, secretARN := range service.Labels {
+			if !strings.HasPrefix(label, converter.SecretLabelPrefix) {
+				continue
+			}
+			namespaces := strings.Split(label, ".")
+			secretName := namespaces[len(namespaces)-1]
+
+			containerSecrets = append(containerSecrets, secrets.NewContainerSecret(service.Name, secretName, secretARN))
+		}
+	}
+	return containerSecrets
+}
+
+func decryptSecrets(containerSecrets []*secrets.ContainerSecret) (envVars map[string]string) {
+	ssmClient, err := newSSMDecrypter()
+	secretsManagerClient, err := newSecretsManagerDecrypter()
+	if err != nil {
+		logrus.Fatal(err.Error())
+	}
+
+	envVars = make(map[string]string)
+	for _, containerSecret := range containerSecrets {
+		service, err := containerSecret.ServiceName()
+		if err != nil {
+			logrus.Fatalf("Failed to retrieve the service of the secret due to %v", err)
+		}
+
+		decrypted := ""
```

Comment:
Would this mean that there could be an empty string in the map of env vars?  Also My spider senses are tingling a little bit here for a way to refactor this (using a visitor pattern or something, eg), though it's readable enough as is.